client/src/components/helpers/AvatarRenderer.js

```diff
@@ -15,9 +15,23 @@ const propTypes = {
 
 function AvatarRenderer({ picture, userName, isDonating, isTopContributor }) {
   const { t } = useTranslation();
+  const [isPictureValid, setIsPictureValid] = useState(true);
   let borderColor = borderColorPicker(isDonating, isTopContributor);
-  let isPlaceHolderImage =
-    /example.com|identicon.org/.test(picture) || picture === defaultUserImage;
+  const onImageLoad = () => setIsPictureValid(true);
+  const onImageError = () => setIsPictureValid(false);
+  let validationImage = document.createElement('img');
+  validationImage.src = picture;
+
+  useEffect(() => {
+    validationImage.onload = onImageLoad;
+    validationImage.onerror = onImageError;
+    // eslint-disable-next-line
```

Comment:
I had here a lint error saying that I should add validationImage.onload as a dependency to useEffect. I am open for any better suggestion

Reply:
I *think* this is fine and that eslint is just getting confused.

Reply:
To be honest, I am not sure I see the reason behind having this within a `useEffect`.

As the definition of the function does not change, surely we want to define the `.on...` callbacks the same time as the HTMLImgElement?

Reply:
Not sure I got what you meant @ShaunSHamilton, can you please explain it in another way ?

Reply:
`useEffect` is used, when you want some piece of code to run at a specific time. Currently, every time `picture` changes, the `onload` and `onerror` functions are redeclared. However, as neither the `onImageLoad` nor `onImageError` function definitions change based on any external props or state, this block of code is doing nothing but causing a performance penalty.

It is equivalent to:
```javascript
const myObj = { onload: null };
function myFunc() { return 1 }
for (let a = 0; a < 10; a++) {
  // Where 10 could be any number based on the
  // lifetime of the component
  myObj.onload = myFunc;
}
```

I hope this is clearer.

Reply:
How about now ? @ShaunSHamilton

Reply:
Now, every time `picture` changes, `isPictureValid` is being set to `true`, then immediately set to `false`.